pkg/controller/webhook/kubefedconfig/webhook.go

```diff
@@ -83,32 +89,68 @@ func (a *KubeFedConfigAdmissionHook) Validate(admissionSpec *admissionv1beta1.Ad
 var _ apiserver.MutatingAdmissionHook = &KubeFedConfigAdmissionHook{}
 
 func (a *KubeFedConfigAdmissionHook) MutatingResource() (plural schema.GroupVersionResource, singular string) {
-	klog.Infof("New MutatingResource for %q", resourceName)
-	return webhook.NewMutatingResource(resourcePluralName), strings.ToLower(resourceName)
+	klog.Infof("New MutatingResource for %q", ResourceName)
+	return webhook.NewMutatingResource(resourcePluralName), strings.ToLower(ResourceName)
 }
 
 func (a *KubeFedConfigAdmissionHook) Admit(admissionSpec *admissionv1beta1.AdmissionRequest) *admissionv1beta1.AdmissionResponse {
 	status := &admissionv1beta1.AdmissionResponse{}
-	klog.V(4).Infof("Admitting %q AdmissionRequest = %s", resourceName, webhook.AdmissionRequestDebugString(admissionSpec))
+	klog.V(4).Infof("Admitting %q AdmissionRequest = %s", ResourceName, webhook.AdmissionRequestDebugString(admissionSpec))
 
 	admittingObject := &v1beta1.KubeFedConfig{}
 	err := webhook.Unmarshal(admissionSpec, admittingObject, status)
 	if err != nil {
 		return status
 	}
 
-	klog.V(4).Infof("Admitting %q = %+v", resourceName, *admittingObject)
+	klog.V(4).Infof("Admitting %q = %+v", ResourceName, *admittingObject)
 
 	if !webhook.Initialized(&a.initialized, &a.lock, status) {
 		return status
 	}
 
-	// TODO(font) add defaults
+	defaultedObject := admittingObject.DeepCopyObject().(*v1beta1.KubeFedConfig)
+	defaults.SetDefaultKubeFedConfig(defaultedObject)
 
+	if reflect.DeepEqual(admittingObject, defaultedObject) {
+		status.Allowed = true
+		return status
+	}
+
+	// TODO(font) Optimize by generalizing the ability to add only the fields
+	// that have been defaulted. If merge patch is ever supported use that
+	// instead.
+	patchOperations := []patchOperation{
```

Comment:
Apologies in advance for my ignorance. Is a mutating webhook called on both create and update?  I assumed that it would only be on create, but the use of a patch operation would seem to suggest otherwise.  Is this the canonical way to do it vs just updating the resource that gets passed in?

Reply:
Hmm, that's interesting.  I was assuming direct mutation of a request (ala a python wsgi pipeline).  I'm curious what prompted this way of doing things.